grouper/fe/handlers.py

```diff
@@ -717,6 +718,136 @@ def get(self, group_id=None, name=None):
         )
 
 
+class AuditsComplete(GrouperHandler):
+    def post(self, audit_id):
+        # TODO: Make it so only certain users can start audits
+        user = self.get_current_user()
+
+        audit = self.session.query(Audit).filter(Audit.id == audit_id).one()
+        if audit.complete:
+            return self.redirect("/groups/{}".format(audit.group.name))
+
+        edges = {}
+        for argument in self.request.arguments:
+            if argument.startswith('audit_'):
+                edges[int(argument.split('_')[1])] = self.request.arguments[argument][0]
+
+        for member in audit.my_members():
+            if member.id in edges:
+                if edges[member.id] in AUDIT_STATUS_CHOICES:
+                    member.status = edges[member.id]
+
+        self.session.commit()
+
+        # Now if it's completable (no pendings) then mark it complete, else redirect them
+        # to the group page.
+        if not audit.completable:
+            return self.redirect('/groups/{}'.format(audit.group.name))
+
+        # Complete audits have to be "enacted" now. This means anybody marked as remove has to
+        # be removed from the group now.
+        for member in audit.my_members():
+            if member.status == "remove":
+                audit.group.revoke_member(self.current_user, member.member,
+                                          "Revoked as part of audit.")
+                AuditLog.log(self.session, self.current_user.id, 'remove_member',
+                             'Removed membership in audit: {}'.format(member.member.name),
+                             on_group_id=audit.group.id)
+
+        audit.complete = True
+        self.session.commit()
+
+        AuditLog.log(self.session, self.current_user.id, 'complete_audit',
+                     'Completed group audit.', on_group_id=audit.group.id)
+
+        return self.redirect('/groups/{}'.format(audit.group.name))
+
+
+class AuditsCreate(GrouperHandler):
+    def get(self):
+        self.render(
+            "audit-create.html", form=AuditCreateForm(),
+        )
+
+    def post(self):
+        form = AuditCreateForm(self.request.arguments)
+        if not form.validate():
+            return self.render(
+                "audit-create.html", form=form,
+                alerts=self.get_form_alerts(form.errors)
+            )
+
+        # TODO: Make it so only certain users can start audits
+        user = self.get_current_user()
+
+        # Step 1, detect if there are non-completed audits and close them.
+        open_audits = self.session.query(Audit).filter(
+            Audit.complete == False).all()
+        for audit in open_audits:
+            audit.complete = True
```

Comment:
If I understand correctly, this is recording that "the last round of group audits finished normally" even though there may be pending and/or unactioned statuses.  Should we record something more detailed about the status of such a past group audits?